src/allotropy/parsers/revvity_kaleido/kaleido_structure.py

```diff
@@ -0,0 +1,410 @@
+        except AllotropeConversionError:
+            logging.warning(
+                f"Unable to read analysis result '{analysis_parameter}'. Ignoring"
+            )
+            return None
+
+        for column in results:
+            if str(column).startswith("Unnamed"):
+                results = results.drop(columns=column)
+
+        try:
+            a1_str_value = str(results.iloc[0, 0])
+        except IndexError as e:
+            error = f"Unable to find first position of analysis result '{analysis_parameter}'."
+            raise AllotropeConversionError(error) from e
+
+        if try_float_or_none(a1_str_value) is None:
+            return None
+
+        return AnalysisResult(
+            analysis_parameter=analysis_parameter,
+            results=results,
+        )
+
+    def get_image_feature_name(self) -> str:
+        return self.analysis_parameter
+
+    def get_result(self, well_position: WellPosition) -> str:
+        try:
+            return str(self.results.loc[well_position.row, well_position.column])
+        except KeyError as e:
+            error = f"Unable to get well at position '{well_position}' from analysis result '{self.analysis_parameter}'."
+            raise AllotropeConversionError(error) from e
+
+    def get_image_feature_result(self, well_position: WellPosition) -> float:
+        return try_float(
+            self.get_result(well_position),
+            f"analysis result '{self.analysis_parameter}' at '{well_position}'",
+        )
+
+
+@dataclass(frozen=True)
+class MeasurementInfo:
+    elements: dict[str, str]
+
+    def get_instrument_serial_number(self) -> str:
+        return assert_not_none(
+            self.elements.get("Instrument Serial Number"),
+            msg="Unable to find Instrument Serial Number in Measurement Information section.",
+        )
+
+    def get_measurement_time(self) -> str:
+        return assert_not_none(
+            self.elements.get("Measurement Started"),
+            msg="Unable to find Measurement time in Measurement Information section.",
+        )
+
+    def get_protocol_signature(self) -> str:
+        return assert_not_none(
+            self.elements.get("Protocol Signature"),
+            msg="Unable to find Protocol Signature in Measurement Information section.",
+        )
+
+    def get_measurement_signature(self) -> str:
+        return assert_not_none(
+            self.elements.get("Measurement Signature"),
+            msg="Unable to find Measurement Signature in Measurement Information section.",
+        )
+
+
+@dataclass(frozen=True)
+class PlateType:
+    elements: dict[str, str]
+
+    @staticmethod
+    def create(reader: CsvReader) -> PlateType:
+        assert_not_none(
+            reader.drop_until_inclusive("^Plate Type"),
+            msg="Unable to find Plate Type section.",
+        )
+        reader.drop_until("^Platemap")
+        return PlateType({})
+
+
+@dataclass(frozen=True)
+class Platemap:
+    data: pd.DataFrame
```

Comment:
Same comment as above. For this one, see perkin elmer envision plate map for an example.